### DISCUSSION TITLE
Opt-in blur for transparent windows

### DISCUSSION THREAD

**Discussion by zopieux** [yshui/picom#731] [Q&A] [answered]
Hi,

_tl;dr does anyone know how to disable background blurring for everything *except* some list?_

The only use I have for background-blurred semi-transparent windows is my terminal emulator. I am therefore trying to configure picom to do the following:

* detect transparent windows and apply opacity; this works fine today
* do not background-blur anything
* in the specific case of a `class_g = 'konsole'` or `name = 'konsole'` (something like that), apply background-blurring

picom config has a `blur-background-exclude` directive, but as the name implies, it's a blocklist/opt-out. I would like it to be opt-in/allowlist instead, with just my terminal emulator selector(s) in that allowlist.

Does anyone have a tip? Thanks a lot!

**Comment by vincentbernat** [comment, marked as answer]
```
blur-background-exclude = [
  "class_g != 'konsole' &&
   window_type != 'tooltip' &&
   window_type != 'menu' &&
   window_type != 'popup_menu' &&
   window_type != 'dropdown_menu' &&
   window_type != 'splash' &&
   window_type != 'combo'
];
```

(I have added more opt-in to show how to extend)

**Reply by Flashwalker** [reply]
> ```
> blur-background-exclude = [
>   "class_g != 'konsole' &&
>   ...
>    window_type != 'combo'" # <- HERE
> ];
> ```
> 
> (I have added more opt-in to show how to extend)

closing double-quote is missing in you listing

**Comment by zopieux** [comment]
Well I feel dumb now, did not realize this had more generic boolean logic capabilities. If I failed to find that in the docs, sorry. Otherwise this probably wouldn't hurt to add something about that "DSL". Thanks a lot!
